### PR TITLE
feat: use same initial filter `owned` at TechDocsIndexPage as at CatalogPage

### PR DESCRIPTION
**Pull request by pjungermann**
Aligns filter use of catalog page (and api page) with techdocs page.

Signed-off-by: Patrick Jungermann <Patrick.Jungermann@gmail.com>

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/wicked-dragons-brush.md
Comment (Rugvip) [inline .changeset/wicked-dragons-brush.md:5]:
```suggestion
Use same initial filter `owned` for the `TechDocsIndexPage` as for the `CatalogPage`.
```

**Review by Rugvip — APPROVED** [review]
Thanks! 👍 

Approved to unblock changeset, leaving the rest to @backstage/techdocs-core

**Comment by camilaibs** [comment]
> Thanks! 👍
> 
> Approved to unblock changeset, leaving the rest to @backstage/techdocs-core

Starting to review!

**Comment by benjdlambert** [comment]
Looks like the `api-reports` need regenerating too here.

**Comment by pjungermann** [comment]
> Looks like the `api-reports` need regenerating too here.

Fixed

File: plugins/techdocs/src/home/components/TechDocsIndexPage.tsx
Comment (freben) [inline plugins/techdocs/src/home/components/TechDocsIndexPage.tsx:24]: This perhaps suggests that the props should have been named `TechDocsIndexPageProps`. Could have renamed and marked the old one as deprecated, if you wanted to.

**Review by freben — COMMENTED** [review]
Had a comment, but dropping my assignment to @backstage/techdocs-core

File: plugins/techdocs/src/home/components/TechDocsIndexPage.tsx
Comment (emmaindal) [reply plugins/techdocs/src/home/components/TechDocsIndexPage.tsx:24]: Yes I think you have a good point here, @pjungermann can we rename `DefaultTechDocsHomeProps` to `TechDocsIndexPageProps` and continue to export that + mark `DefaultTechDocsHomeProps` as deprecated as part of this PR. That way one can use `TechDocsIndexPageProps` for any custom homepage if wanted and it will be a bit more generic.

**Review by emmaindal — COMMENTED** [review]
Nice! I've just a small additional comment to what @freben suggested, if we can get that in as part of this PR it would be nice!

File: plugins/techdocs/src/home/components/TechDocsIndexPage.tsx
Comment (pjungermann) [reply plugins/techdocs/src/home/components/TechDocsIndexPage.tsx:24]: @freben @emmaindal 
I followed the same as at CatalogPage

```
export function CatalogPage(props: DefaultCatalogPageProps) {
  const outlet = useOutlet();

  return outlet || <DefaultCatalogPage {...props} />;
}
```

but if you prefer I can change it.

**Comment by iamEAP** [comment]
Hmm...  If I step _really_ far back, I wonder if we actually _do_ want to align the default filter values, here?

If I visit the Catalog homepage, it seems (slightly) more likely that I would want to see stuff that I own because I am in "manage" mode.  I need to perform some task related to some piece of software that I work with regularly.

But if I'm on the _TechDocs_ homepage, it's less likely that I want it pre-filtered to things I own: I don't need the docs for things I work with regularly, it's all in my head already.  Instead, I'm more in "explore" mode, looking for docs related to software I'm less familiar with / probably don't own.

Does that resonate with you @pjungermann?  Or what was your motivation in aligning these filters?

\cc @garyniemen @soapraj

File: .changeset/wicked-dragons-brush.md
Comment (iamEAP) [inline .changeset/wicked-dragons-brush.md:5]: Maybe a note in here about "if you'd prefer to keep the default filter to `all`, do [x, y, z]`" could be useful?  It'd be nice to at least highlight that it's (more easily?) configurable now.

File: plugins/techdocs/src/home/components/TechDocsIndexPage.tsx
Comment (iamEAP) [reply plugins/techdocs/src/home/components/TechDocsIndexPage.tsx:24]: Will just echo that it'd be awesome to make the change suggested by Emma before we merge (including deprecation of the existing type/name); it'll make it clearer if anyone needs to type a variable that's provided to `<TechDocsIndexPage />` if the name of the type corresponds to the component.

**Review by iamEAP — COMMENTED** [review]
Makes total sense @pjungermann, and agree, what makes the most sense is probably going to depend entirely on the organization's specific uses/needs.

We discussed briefly within the `@backstage/techdocs-core` team internally and think the alignment of the filter values makes sense, especially since they are user-configurable.

With that in mind!  Just one comment on how the changeset can be improved and then just re-upping the suggestion to rename the type + deprecate the old type while you're at it.

Then we can ship this.

**Review by iamEAP — APPROVED** [review]
🏅 Thanks again @pjungermann!

**Pull request merged**
